src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -508,11 +508,11 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                                     {
                                         compilation.ReportUnusedImports(null, diagnosticBag, cancellationToken);
                                     }
-
-                                    compilation.CompleteTrees(null);
                                 }
                             }
 
+                            compilation.CompleteTrees(null);
```

Comment:
In the case of an exception this method won't be called.  This meant the async queue won't follow it's desired state transition.  Will that cause deadlocks / issues down the road?  What are the guarantees we provide?  

@gafter may have input here.